eth_keys/backends/native/ecdsa.py

```diff
@@ -60,6 +60,35 @@ def private_key_to_public_key(private_key_bytes: bytes) -> bytes:
     return public_key_bytes
 
 
+def compress_public_key(uncompressed_public_key_bytes: bytes) -> bytes:
+    x, y = decode_public_key(uncompressed_public_key_bytes)
+    if y % 2 == 0:
+        prefix = b"\x02"
+    else:
+        prefix = b"\x03"
+    return prefix + pad32(int_to_big_endian(x))
+
+
+def decompress_public_key(compressed_public_key_bytes: bytes) -> bytes:
+    if len(compressed_public_key_bytes) != 33:
+        raise ValueError("Invalid compressed public key")
+
+    prefix = compressed_public_key_bytes[0]
+    if prefix not in (2, 3):
+        raise ValueError("Invalid compressed public key")
+
+    x = big_endian_to_int(compressed_public_key_bytes[1:])
```

Comment:
I've been on a big `memoryview` kick recently so feel free to ignore me.  This would be slightly faster using a `memoryview` to prevent the copy.  This is only safe if the thing you're passing it to doesn't try to mutate it which in this case I believe that condition is satisfied.

Reply:
I did a quick benchmark and didn't see any improvement. I think the reason is that it would save one copy when slicing, but it adds one when creating the `memoryview`. But I will keep it in mind in the future (could be very useful for pyrlp/ssz).